pulp_ansible/app/tasks/copy.py

```diff
@@ -62,3 +73,80 @@ def process_entry(entry):
         base_version = dest_repo_version if dest_version_provided else None
         with dest_repo.new_version(base_version=base_version) as new_version:
             new_version.add_content(content_to_copy)
+
+
+def copy_collection(cv_pk_list, src_repo_pk, dest_repo_list):
+    """
+    Copy a collection and all of it's contents into a list of destination repositories
+    """
+    src_repo = AnsibleRepository.objects.get(pk=src_repo_pk)
+    collection_versions = CollectionVersion.objects.filter(pk__in=cv_pk_list)
+
+    content_types = src_repo.content.values_list("pulp_type", flat=True).distinct()
+    source_pks = src_repo.content.values_list("pk", flat=True)
+
+    content = cv_pk_list
+
+    # collection signatures
+    if "ansible.collection_signature" in content_types:
+        signatures_pks = CollectionVersionSignature.objects.filter(
+            signed_collection__in=cv_pk_list, pk__in=source_pks
+        ).values_list("pk", flat=True)
+        if signatures_pks:
+            content.extend(signatures_pks)
+
+    # collection version mark
+    if "ansible.collection_mark" in content_types:
+        marks_pks = CollectionVersionMark.objects.filter(
+            marked_collection__in=cv_pk_list, pk__in=source_pks
+        ).values_list("pk", flat=True)
+        if marks_pks:
+            content.extend(marks_pks)
+
+    # namespace metadata
+    namespaces = {x.namespace for x in collection_versions}
+    namespaces_pks = AnsibleNamespaceMetadata.objects.filter(
+        pk__in=source_pks,
+        name__in=list(namespaces),
+    ).values_list("pk", flat=True)
+    if namespaces_pks:
+        content.extend(namespaces_pks)
+
+    # collection deprecation
+    for cv in collection_versions:
+        if "ansible.collection_deprecation" in content_types:
+            deprecations_pks = AnsibleCollectionDeprecated.objects.filter(
+                pk__in=source_pks,
+                namespace=cv.namespace,
+                name=cv.name,
+            ).values_list("pk", flat=True)
+            if deprecations_pks:
+                content.extend(deprecations_pks)
+
+    for pk in dest_repo_list:
+        add_and_remove(repository_pk=pk, add_content_units=content, remove_content_units=[])
+
+
+def move_collection(cv_pk_list, src_repo_pk, dest_repo_list):
+    """
+    Copy a collection and all of it's contents into a list of destination repositories
+    """
+    copy_collection(cv_pk_list, src_repo_pk, dest_repo_list)
+
+    # don't need to clean anything other than the collection version because everything
+    # else will get handled by AnsibleRepository.finalize_repo_version()
```

Comment:
```suggestion
    # No need to clean anything other than the collection version because everything
    # else will get handled by AnsibleRepository.finalize_repo_version().
```